tangelo/linq/helpers/circuits/clifford_circuits.py

```diff
@@ -0,0 +1,71 @@
+from math import pi, isclose
+
+from tangelo.linq import Gate
+
+
+def decompose_gate_to_cliffords(gate, abs_tol=1e-4):
+    """
+    Decomposes a single qubit parameterized gate into Clifford gates.
+
+    Parameters:
+    gate (Gate): The gate to be decomposed.
+    abs_tol (float): Optional, absolute tolerance for value comparison (default: 1e-4).
+
+    Returns:
+    list: A list of Clifford gates representing the decomposition.
+
+    Raises:
+    ValueError: If parameterized gate cannot be decomposed into Clifford gates.
+
+    """
+
+    if gate.name not in {"RX", "RY", "RZ", "PHASE"}:
+        return gate
+    gate_list = []
+    clifford_values = [0, pi, pi / 2, -pi / 2]
+    value_isclose = [isclose(gate.parameter, value, abs_tol=abs_tol) for value in clifford_values]
+
+    if not any(value_isclose):
+        raise ValueError(
```

Comment:
Is there a situation where one could maybe loop over all gates of a circuit and try to decompose it into Clifford gates as much as possible but be ok with having some non-clifford gates in there ? Or do you think the only purpose here is to decompose into Clifford and it should fail and return an error if we can't ?